process/transaction/interceptedTransaction.go

```diff
@@ -165,12 +186,49 @@ func (inTx *InterceptedTransaction) CheckValidity() error {
 			return err
 		}
 
+		err = inTx.verifyIfRelayedTxV2(inTx.tx)
+		if err != nil {
+			return err
+		}
+
 		inTx.whiteListerVerifiedTxs.Add([][]byte{inTx.Hash()})
 	}
 
 	return nil
 }
 
+func (inTx *InterceptedTransaction) verifyIfRelayedTxV2(tx *transaction.Transaction) error {
+	funcName, userTxArgs, err := inTx.argsParser.ParseCallData(string(tx.Data))
+	if err != nil {
+		return nil
+	}
+	if core.RelayedTransactionV2 != funcName {
+		return nil
+	}
+
+	userTx, err := createRelayedV2(tx, userTxArgs)
+	if err != nil {
+		return err
+	}
+
+	err = inTx.verifySig(userTx)
+	if err != nil {
+		return err
+	}
+
+	funcName, _, err = inTx.argsParser.ParseCallData(string(userTx.Data))
+	if err != nil {
+		return nil
+	}
+
+	// recursive relayed transactions are not allowed
+	if core.RelayedTransaction == funcName || core.RelayedTransactionV2 == funcName {
```

Comment:
might extract in a new function, just like you did in the block/preprocess/gascomputation.go

Reply:
done